bot/cogs/moderation/infractions.py

```diff
@@ -244,7 +247,14 @@ def __init__(self, bot: Bot):
 
         self.mod_log.ignore(Event.member_remove, user.id)
 
-        action = ctx.guild.ban(user, reason=reason, delete_message_days=0)
+        if len(reason) > 512:
+            log.info("Ban reason is longer than 512 characters. Reason will be truncated for Audit Log.")
+
+        action = ctx.guild.ban(
+            user,
+            reason=f"{reason[:509]}..." if len(reason) > 512 else reason,
+            delete_message_days=0
+        )
```

Comment:
Should we put this in a try except and move it before we call `utils.post_infraction()`? It will be rare, but there can be situation where the actual discord ban / kick failed, but since we've already put the infraction in the DB the bot will think that the ban is active.

Reply:
Should this after coroutine making or after `apply_infraction`?

Reply:
Also, I can't move to add into DB to end of the function: `infraction` is required for applying infraction, and in coroutine creation, this will not raise an error. An error will be raised on `self.apply_infraction`, but this need DB returned infraction.